File: S4-Systems-for-Design-and-Impl/Labs/Lab7/ProiectJava/TransportNetworking/src/main/java/transport/network/utils/TransportRpcConcurrentServer.java
package transport.network.utils;

import transport.network.rpcprotocol.TransportClientRpcReflectionWorker;
import transport.services.ITransportServices;

import java.net.Socket;

public class TransportRpcConcurrentServer extends AbsConcurrentServer {
    private ITransportServices transportServer;
    public TransportRpcConcurrentServer(int port, ITransportServices transportServer) {
        super(port);
        this.transportServer = transportServer;
        System.out.println("Transport - TransportRpcConcurrentServer");
    }

    @Override
    protected Thread createWorker(Socket client) {
        TransportClientRpcReflectionWorker worker = new TransportClientRpcReflectionWorker(transportServer, client);

        Thread tw = new Thread(worker);
        return tw;
    }

    @Override
    public void stop(){
        System.out.println("Stopping services ...");
    }
}

File: Systems-for-Design-and-Implementation/Labs/Lab8/ProiectCSharp/TransportNetworking/transport.network/utils/TransportRpcConcurrentServer.cs
using System;
using System.Net.Sockets;
using System.Threading;
using TransportNetworking.transport.network.jsonprotocol;
using TransportServices.transport.services;

namespace TransportNetworking.transport.network.utils
{
    public class TransportRpcConcurrentServer : AbsConcurrentServer
    {
        private ITransportServices transportServer;

        public TransportRpcConcurrentServer(int port, ITransportServices transportServer) : base(port)
        {
            this.transportServer = transportServer;
            Console.WriteLine("Transport - TransportRpcConcurrentServer");
        }

        protected override Thread CreateWorker(Socket client)
        {
            //TransportClientRpcReflectionWorker worker = new TransportClientRpcReflectionWorker(transportServer, client);
            TransportClientJsonWorker worker = new TransportClientJsonWorker(transportServer, client);

            Thread tw = new Thread(worker.Run);
            return tw;
        }

        public override void Stop()
        {
            Console.WriteLine("Stopping services ...");
            base.Stop();
        }
    }
}